eclair-core/src/test/scala/fr/acinq/eclair/router/ChannelRangeQueriesSpec.scala

```diff
@@ -356,4 +360,37 @@ class ChannelRangeQueriesSpec extends FunSuite {
       validateChunks(chunks.toList, pruned)
     }
   }
+
+  test("compute extended query information") {
+    import QueryFlags._
+
+    val ids = makeShortChannelIds(42, 100) ++ makeShortChannelIds(43, 70) ++ makeShortChannelIds(44, 50) ++ makeShortChannelIds(45, 30) ++ makeShortChannelIds(50, 120)
+    val channels = SortedMap.empty[ShortChannelId, PublicChannel] ++ ids.map(id => id -> RoutingSyncSpec.makeFakeRoutingInfo(id)._1).toMap
+
+    def validate(timestamps: Option[ReplyChannelRangeTlv.EncodedTimestamps], checksums: Option[ReplyChannelRangeTlv.EncodedChecksums], ids: List[ShortChannelId], queryFlags_opt: Option[QueryFlags], encodingType: EncodingType) = {
+      queryFlags_opt match {
+        case None =>
+          require(timestamps.isEmpty)
+          require(checksums.isEmpty)
+        case _ if ids.isEmpty =>
+          require(timestamps.isEmpty)
+          require(checksums.isEmpty)
+        case Some(extension) =>
+          if (extension.wantTimestamps) {
+            require(timestamps.map(_.timestamps.size) == Some(ids.size))
+            require(timestamps.map(_.encoding) == Some(encodingType))
+          }
+          if (extension.wantChecksums) require(checksums.map(_.checksums.size) == Some(ids.size))
```

Comment:
Maybe also check that if `!extension.wantTimestamps` timestamps aren't included, and if `!extension.wantChecksums` checksums aren't included?